### PR TITLE
Fix FAT16 file system issue

### PR DESCRIPTION
**Pull request by henrygab**
Code changes are based on those I made to the adafruit nrf52840 bootloader, which I built and ran under SES's debugger.

This fixes enumeration of files under DOS / CMD windows.

For the most part, this PR adds only static compile-time changes
and comments that warn future maintainers about unwritten
rules and presumptions.

Code change summary:
1. Set the PhysicalDriveNum in the boot block
2. Files get a valid creation and update date

### CODE REVIEW

**Comment by henrygab** [comment]
@mmoskal Are you taking PRs for UF2 still?

File: src/ghostfat.c [hunk 1]
```diff
@@ -94,19 +104,25 @@ static const struct TextFile info[] = {
...
+
 static const FAT_BootBlock BootBlock = {
     .JumpInstruction = {0xeb, 0x3c, 0x90},
     .OEMInfo = "UF2 UF2 ",
     .SectorSize = 512,
     .SectorsPerCluster = 1,
     .ReservedSectors = RESERVED_SECTORS,
     .FATCopies = 2,
-    .RootDirectoryEntries = (ROOT_DIR_SECTORS * 512 / 32),
+    .RootDirectoryEntries = (ROOT_DIR_SECTORS * 512 / DIRENTRIES_PER_SECTOR),
```
Comment (mmoskal) [inline src/ghostfat.c]:
```suggestion
    .RootDirectoryEntries = (ROOT_DIR_SECTORS * DIRENTRIES_PER_SECTOR),
```

The computation was wrong causing trouble in samd21 bootloader

**Review by mmoskal — COMMENTED** [review]
This project was never fully functional - just use the Adafruit's one.

**Comment by henrygab** [comment]
@mmoskal Good catch, thanks.  I also checked my PR to the AdaFruit branch, and it didn't have this issue.

**Pull request closed without merging**